src/__tests__/identify.test.ts

```diff
@@ -0,0 +1,98 @@
+import NotificationAPIClient from '..';
+import { NotificationAPIClientInterface } from '../interfaces';
+
+// Mocking the fetch API
+global.fetch = jest.fn();
+const fetchMock = global.fetch as jest.Mock;
+
+const clientId = 'env:Id';
+const userId = 'user:Id';
+
+let spy: jest.SpyInstance;
+let notificationapi: NotificationAPIClientInterface;
+
+beforeEach(() => {
+  spy = jest.spyOn(console, 'error').mockImplementation();
+  notificationapi = new NotificationAPIClient({
+    clientId,
+    userId
+  });
+});
+
+afterEach(() => {
+  spy.mockRestore();
+  if (notificationapi) notificationapi.destroy();
+});
+
+describe('when identify is called', () => {
+  it('returns error when userIds do not match', async () => {
+    notificationapi.identify({
+      id: 'somethingelse'
+    });
+
+    expect(spy.mock.calls).toEqual([
+      [
+        'The userId "somethingelse" does not match the userId "user:Id" provided in the init options. Cancelling action to prevent mistakes.'
+      ]
+    ]);
+  });
+
```

Comment:
Nest this tests in a describe, for example:
  describe('API request attributes', () => {